### PR TITLE
Feature/xcode 8 support

### PR DESCRIPTION
**Pull request by jboeijenga**
- support xcode 8
- remove no used features
- add demo

### CODE REVIEW

File: demo/app/App_Resources/Android/app.gradle [hunk 1]
```diff
@@ -3,4 +3,21 @@
...
\ No newline at end of file
+//}
+
+android {  
+  defaultConfig {  
+    generatedDensities = []
+    applicationId = "org.nativescript.demo"
+    
+    //override supported platforms
+    // ndk {
```
Comment (spike1292) [inline demo/app/App_Resources/Android/app.gradle:14]: remove commented code?

File: package.json [hunk 2]
```diff
@@ -1,58 +1,32 @@
...
+    "preparedemo": "npm run build && cd demo && tns plugin remove ns-pdfviewer && tns plugin add .. && tns install",
     "setup": "npm install && cd demo && npm install && cd .. && npm run build && cd demo && tns plugin add .. && cd .."
   },
-  "release": {
-    "generateNotes": "./node_modules/@semantic-release/release-notes-generator",
-    "verifyConditions": [
-      "./node_modules/@semantic-release/condition-codeship"
-    ]
-  },
-  "repository": {
```
Comment (spike1292) [inline package.json:28]: pls add needed for npm

File: package.json [hunk 3]
```diff
@@ -1,58 +1,32 @@
 {
-  "name": "nativescript-pdf-view",
-  "description": "A NativeScript plugin to display PDF files on iOS and Android",
-  "main": "src/plugin.js",
-  "version": "1.2.1",
+  "name": "ns-pdfviewer",
+  "version": "1.0.0",
+  "description": "",
+  "main": "pdfviewer.js",
```
Comment (spike1292) [inline package.json]: remove the js extension, nativescript doesn't like it

File: package.json [hunk 4]
```diff
@@ -1,58 +1,32 @@
 {
-  "name": "nativescript-pdf-view",
-  "description": "A NativeScript plugin to display PDF files on iOS and Android",
-  "main": "src/plugin.js",
-  "version": "1.2.1",
+  "name": "ns-pdfviewer",
```
Comment (spike1292) [inline package.json]: think the old name is good, but then we need to come up with a beter package name for example`nativescript-pdf-viewer-xcode-8`

File: package.json [hunk 5]
```diff
@@ -1,58 +1,32 @@
...
-    "generateNotes": "./node_modules/@semantic-release/release-notes-generator",
-    "verifyConditions": [
-      "./node_modules/@semantic-release/condition-codeship"
-    ]
-  },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/Merott/nativescript-pdf-view.git"
-  },
-  "keywords": [
```
Comment (spike1292) [inline package.json:32]: pls add needed for npm

File: pdfviewer.common.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+			viewer.notify({ eventName, object: viewer });
+		}
+	}
+
+	public get src(): string {
+		return this._getValue(PdfViewer.srcProperty);
+	}
+
+	public set src(src: string) {
+		console.log('pdf viewer load url: ' + src);
```
Comment (spike1292) [inline pdfviewer.common.ts]: console.log?

File: pdfviewer.ios.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+		this.ios.autoresizingMask =
+			UIViewAutoresizing.UIViewAutoresizingFlexibleWidth |
+			UIViewAutoresizing.UIViewAutoresizingFlexibleHeight;
+
+		this.ios.scalesPageToFit = true;
+		this.ios.scrollView.decelerationRate = UIScrollViewDecelerationRateNormal
+	}
+}
+
+function iosProperty(theClass, theProperty) {
```
Comment (spike1292) [inline pdfviewer.ios.ts]: 👍

File: src/plugin.common.ts [hunk 8]
```diff
@@ -1,40 +0,0 @@
...
-
-  public set src(src: string) {
-    this._setValue(PDFView.srcProperty, src);
-    this.loadPDF(src);
-  }
-
-  public abstract loadPDF(src: string);
-}
-
-try {
```
Comment (spike1292) [inline src/plugin.common.ts:33]: why is this removed?

File: index.d.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+/**
+ * iOS and Android apis should match.
+ * It doesn't matter if you export `.ios` or `.android`, either one but only one.
+ */
+export * from './pdfviewer.common';
```
Comment (spike1292) [inline index.d.ts]: add 
`export * from './pdfviewer.ios'`

File: pdfviewer.common.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import { Property } from 'ui/core/dependency-observable';
+import dependencyObservable = require("ui/core/dependency-observable");
+import * as proxy from 'ui/core/proxy';
+import * as view from 'ui/core/view';
+
+export abstract class PdfViewer extends view.View {
+	private static srcProperty = new Property('src', 'PdfViewer', new proxy.PropertyMetadata(''));
+	public static loadEvent = 'load';
+
+	public static notifyOfEvent(eventName: string, pdfViewRef: WeakRef<PdfViewer>) {
```
Comment (spike1292) [inline pdfviewer.common.ts]: not used so can it be removed?

File: platforms/android/include.gradle [hunk 11]
```diff
@@ -1,7 +1,7 @@
 android {
   productFlavors {
-    "nativescript-pdf-view" {
-      dimension "nativescript-pdf-view"
+    "ns-pdfviewer" {
```
Comment (spike1292) [inline platforms/android/include.gradle]: if the packagename is changed, dont forget this

**Review by spike1292 — CHANGES_REQUESTED** [review]

**Review by dennis-montana — APPROVED** [review]

File: demo/package.json [hunk 12]
```diff
@@ -1,31 +1,41 @@
...
+    "@angular/common": "2.1.2",
+    "@angular/compiler": "2.1.2",
+    "@angular/core": "2.1.2",
+    "@angular/forms": "2.1.2",
+    "@angular/http": "2.1.2",
+    "@angular/platform-browser": "2.1.2",
+    "@angular/platform-browser-dynamic": "2.1.2",
+    "@angular/router": "3.1.2",
+    "nativescript-angular": "1.1.3",
+    "nativescript-pdf-viewer": "file:///Users/Henk/Projects/nativescript/nativescript-pdf-view",
```
Comment (spike1292) [inline demo/package.json:25]: remove

**Review by spike1292 — APPROVED** [review]

**Pull request merged**